java/core/src/main/java/com/google/protobuf/Utf8.java

```diff
@@ -613,8 +614,31 @@ private static int partialIsValidUtf8(final ByteBuffer buffer, int index, final
      *
      * @throws InvalidProtocolBufferException if the byte array slice is not valid UTF-8.
      */
-    abstract String decodeUtf8(byte[] bytes, int index, int size)
-        throws InvalidProtocolBufferException;
+    String decodeUtf8(byte[] bytes, int index, int size)
```

Comment:
@elharo should I replace the body of the existing

```
  static String decodeUtf8(byte[] bytes, int index, int size)
      throws InvalidProtocolBufferException {
    return processor.decodeUtf8(bytes, index, size);
  }

```

with this implementation and remove

`String decodeUtf8(byte[] bytes, int index, int size)`

from `Utf8::Processor`?

Reply:
I think you can simply call this without copying and pasting

Reply:
@elharo not only copying and pasting, also removing `String decodeUtf8(byte[] bytes, int index, int size)` from `Utf8::Processor'.

There's no longer a point in keeping the implementation in `Processor` since there are no safe and unsafe implementations, only a safe one.

Reply:
I pushed this change to make the intent clear.